### PR TITLE
Use `aarch64` artifact arch for integration tests on Mac M1 (darwin) OS.

### LINKED ISSUE #14753: Running integration test on Mac M1, `fileBeat` artifact is not revealed.
### Issue description
When running integration tests (`ci/integration_tests.sh specs/monitoring_api_spec.rb`) as [guideline states](https://github.com/elastic/logstash/blob/main/qa/integration/README.md#running-integration-tests-locally-maclinux) on local Mac M1 machine the process fails. The log file shows `fileBeat` artifact cannot be found for `filebeat-8.6.0-SNAPSHOT-darwin-arm64.tar.gz`.

```
FAILURE: Build failed with an exception.

* Where:
Build file '/Users/mashhur/logstash/build.gradle' line: 464

* What went wrong:
Execution failed for task ':downloadFilebeat'.
> java.io.FileNotFoundException: https://artifacts-api.elastic.co/v1/versions/8.6.0-SNAPSHOT/builds/8.6.0-9a0f87a9/projects/beats/packages/filebeat-8.6.0-SNAPSHOT-darwin-arm64.tar.gz
```

### Solution
It seems previously `fileBeat` artifact generated differently for `darwin`/Mac but now, as we can check in [artifact repo](https://artifacts-api.elastic.co/v1/versions/8.6.0-SNAPSHOT/builds/8.6.0-9a0f87a9/projects), `darwin-aarch64` is available (`filebeat-8.6.0-SNAPSHOT-darwin-aarch64.tar.gz`). We will use this artifact for Mac M1.

### PR DESCRIPTION
**Pull request by mashhurs**
## Release notes

[rn:skip] 

## What does this PR do?
Fixes the `fileBeat` artifact not found issue when running integration tests on Mac M1.

## Why is it important/What is the impact to the user?
Not impacted.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] ~~I have made corresponding changes to the documentation~~
- [ ] ~~I have made corresponding change to the default configuration files (and/or docker env variables)~~
- [ ] ~~I have added tests that prove my fix is effective or that my feature works~~

## Author's Checklist

- [ ]

## How to test this PR locally
- Pull the changes
- Run `ci/integration_tests.sh specs/monitoring_api_spec.rb` to test single integration test.

## Related issues
- Closes #14753 

## Logs
```
// issue case

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/mashhur/Dev/elastic/logstash/build.gradle' line: 464

* What went wrong:
Execution failed for task ':downloadFilebeat'.
> java.io.FileNotFoundException: https://artifacts-api.elastic.co/v1/versions/8.6.0-SNAPSHOT/builds/8.6.0-9a0f87a9/projects/beats/packages/filebeat-8.6.0-SNAPSHOT-darwin-arm64.tar.gz

```

```
// after fix
> Task :downloadFilebeat
Caching disabled for task ':downloadFilebeat' because:
  Build cache is disabled
Task ':downloadFilebeat' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
Download https://snapshots.elastic.co/8.6.0-9a0f87a9/downloads/beats/filebeat/filebeat-8.6.0-SNAPSHOT-darwin-aarch64.tar.gz
Downloaded to /Users/mashhur/Dev/elastic/logstash/build/filebeat-8.6.0-SNAPSHOT-darwin-aarch64.tar.gz
:downloadFilebeat (Thread[Execution worker Thread 7,5,main]) completed. Took 8.915 secs.
Resolve mutations for :copyFilebeat (Thread[Execution worker Thread 5,5,main]) started.
Resolve mutations for :copyFilebeat (Thread[Execution worker Thread 5,5,main]) completed. Took 0.0 secs.
:copyFilebeat (Thread[Execution worker Thread 7,5,main]) started.

> Task :copyFilebeat
Caching disabled for task ':copyFilebeat' because:
  Build cache is disabled
Task ':copyFilebeat' is not up-to-date because:
  Task has not declared any outputs despite executing actions.
Unzipped /Users/mashhur/Dev/elastic/logstash/build/filebeat-8.6.0-SNAPSHOT-darwin-aarch64.tar.gz to ./build/filebeat
Deleting /Users/mashhur/Dev/elastic/logstash/build/filebeat-8.6.0-SNAPSHOT-darwin-aarch64.tar.gz
:copyFilebeat (Thread[Execution worker Thread 7,5,main]) completed. Took 1.549 secs.
Resolve mutations for :logstash-integration-tests:compileJava (Thread[Execution worker Thread 5,5,main]) started.
Resolve mutations for :logstash-integration-tests:compileJava (Thread[Execution worker Thread 5,5,main]) completed. Took 0.0 secs.
:logstash-integration-tests:compileJava (Thread[Execution worker Thread 7,5,main]) started.
```

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -142,24 +142,19 @@ String artifactVersionsApi = "https://artifacts-api.elastic.co/v1/versions"
 
 tasks.register("configureArchitecture") {
     String arch = System.properties['os.arch']
-    String osName = System.properties['os.name']
     String beatsArch = arch
     String esArch = arch
+    String osName = (System.properties['os.name'] ==~ /Mac OS X/) ? "darwin" : "linux"
```
Comment (andsel) [inline build.gradle:147]: I think with this, it breaks  on Windows

Comment (mashhurs) [reply build.gradle:147]: This line is the simplification of L158~162L and I think we need to separately visit Windows case. I have a doubt that even if we fix OS name, artifact extensions (such as `.msi`) is not properly set in this logic.

Comment (mashhurs) [reply build.gradle:147]: We recommend using Docker on Windows: [guide link](https://github.com/elastic/logstash/blob/main/qa/integration/README.md#testing-on-windows)

Comment (andsel) [reply build.gradle:147]: Now I see, you are right.

**Review by andsel — APPROVED** [review]
LGTM, but can't test locally because on I'm on x86

**Pull request merged**